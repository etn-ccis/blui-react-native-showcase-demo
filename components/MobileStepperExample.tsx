import React, { useCallback, useState } from 'react';
import { Text, View } from 'react-native';
import { MobileStepper, DotStepperVariant } from '@brightlayer-ui/react-native-components';
import { Button, Card, Divider, useTheme } from 'react-native-paper';
import * as Colors from '@brightlayer-ui/colors';

export const MobileStepperExample: React.FC = () => {
    const theme = useTheme();
    const totalSteps = 5;
    const [currentStep, setCurrentStep] = useState(0);
    const [mobileStepperVariant, setMobileStepperVariant] = useState<DotStepperVariant>('dots');

    const updateStep = useCallback(
        (delta: number): void => {
            setCurrentStep(currentStep + delta);
        },
        [currentStep]
    );

    const cycleStepperVariant = (): void => {
        switch (mobileStepperVariant) {
            case 'dots':
                setMobileStepperVariant('text');
                break;
            case 'text':
                setMobileStepperVariant('progress');
                break;
            default:
                setMobileStepperVariant('dots');
                break;
        }
    };

    return (
        <Card style={{ padding: 0, marginTop: 0, marginHorizontal: 10, marginBottom: 10 }}>
            <Button mode="contained" onPress={(): void => cycleStepperVariant()}>
                Cycle Variant
            </Button>
            <Divider />
            <Text style={{ margin: 32, textAlign: 'center', color: theme.colors.primary }}>
                Step #{currentStep + 1}
            </Text>
            <Divider />
            <MobileStepper
                activeStep={currentStep}
                steps={totalSteps}
                leftButton={
                    <View style={{ flex: 1 }}>
                        <Button
                            style={{ width: 100, alignSelf: 'flex-start' }}
                            disabled={currentStep === 0}
                            onPress={(): void => updateStep(-1)}
                            mode="outlined"
                        >
                            Back
                        </Button>
                    </View>
                }
                rightButton={
                    <View style={{ flex: 1 }}>
                        <Button
                            style={{ width: 100, alignSelf: 'flex-end' }}
                            disabled={currentStep === totalSteps - 1}
                            onPress={(): void => updateStep(1)}
                            mode="contained"
                        >
                            Next
                        </Button>
                    </View>
                }
                variant={mobileStepperVariant}
                activeColor={Colors.blue[500]}
            />
        </Card>
    );
};
