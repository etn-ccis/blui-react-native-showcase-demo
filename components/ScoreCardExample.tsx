import React from 'react';
import { Hero, HeroBanner, InfoListItem, ScoreCard } from '@brightlayer-ui/react-native-components';
import { useTheme } from 'react-native-paper';
import { BLUIColors } from '@brightlayer-ui/colors';

export const ScoreCardExample: React.FC = () => {
    const theme = useTheme();
    return (
        <ScoreCard
            headerTitle={'Substation 42'}
            headerSubtitle={'Normal'}
            headerInfo={'42 Devices'}
            actionItems={[{ icon: { name: 'star-outline' } }, { icon: { name: 'more-vert' }, onPress: () => {} }]}
            badgeOffset={-55}
            badge={
                <HeroBanner style={{ flex: 0, minWidth: 80, justifyContent: 'flex-end' }}>
                    <Hero
                        iconBackgroundColor={theme.colors.surface}
                        label={'Score'}
                        iconSize={48}
                        iconColor={BLUIColors.success[50]}
                        ChannelValueProps={{
                            value: 98,
                            units: '/100',
                        }}
                        icon={{ family: 'brightlayer-ui', name: 'grade_a' }}
                    />
                </HeroBanner>
            }
            actionRow={<InfoListItem chevron title={'View More'} hidePadding />}
        />
    );
};
