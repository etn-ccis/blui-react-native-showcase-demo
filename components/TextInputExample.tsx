import React from 'react';
import { View, StyleSheet } from 'react-native';
import { TextInput, useTheme, HelperText, Text } from 'react-native-paper';

export const TextFieldExample: React.FC = () => {
    const theme = useTheme();
    const [normalText, setNormalText] = React.useState('');
    const [errorText, setErrorText] = React.useState('');
    const [disabledText, setDisabledText] = React.useState('');
    const [normalFilledText, setNormalFilledText] = React.useState('Hello');
    const [errorFilledText, setErrorFilledText] = React.useState('Hello');
    const [disabledFilledText, setDisabledFilledText] = React.useState('Hello');
    const [hasError, setHasError] = React.useState(true);

    const styles = StyleSheet.create({
        container: {
            marginTop: 16,
        },
        flatInput: {
            margin: 8,
        },
        outlineInput: {
            margin: 8,
            backgroundColor: 'transparent',
        },
        thinUnderline: {
            borderBottomWidth: 0, // Adjust the thickness as needed
        },
    });

    return (
        <View>
            <View style={styles.container}>
                <Text> Flat TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Flat TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorText(value);
                        setHasError(value.length > 5);
                    }}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Flat TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    value={disabledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledText(value)}
                    disabled
                    style={[styles.flatInput]}
                />
            </View>
            <View style={styles.container}>
                <Text> Flat Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalFilledText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalFilledText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Flat Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorFilledText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorFilledText(value);
                        setHasError(value.length > 4);
                    }}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Flat Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    value={disabledFilledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledFilledText(value)}
                    disabled
                    style={[styles.flatInput]}
                />
            </View>
            <View style={styles.container}>
                <Text> Dense Flat TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalText}
                    dense={true}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Dense Flat TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorText(value);
                        setHasError(value.length > 5);
                    }}
                    dense={true}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Dense Flat TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    value={disabledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledText(value)}
                    disabled
                    dense={true}
                    style={[styles.flatInput]}
                />
            </View>
            <View style={styles.container}>
                <Text> Dense Flat Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalFilledText}
                    dense={true}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalFilledText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Dense Flat Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorFilledText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorFilledText(value);
                        setHasError(value.length > 4);
                    }}
                    dense={true}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Dense Flat Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="flat"
                    value={disabledFilledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledFilledText(value)}
                    disabled
                    dense={true}
                    style={[styles.flatInput]}
                />
            </View>
            <View style={{ marginTop: 30 }}>
                <Text> Normal Outlined TextInput </Text>
                <TextInput
                    label="Normal Outlined TextInput"
                    mode="outlined"
                    value={normalText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    outlineColor={theme.colors.outline}
                    onChangeText={(value) => setNormalText(value)}
                    style={styles.outlineInput}
                />

                <Text> Error Outlined TextInput </Text>
                <TextInput
                    label="Error Outlined TextInput"
                    mode="outlined"
                    value={errorText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    outlineColor={theme.colors.outline}
                    onChangeText={(value) => {
                        setErrorText(value);
                        setHasError(value.length > 5);
                    }}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Outlined TextInput </Text>
                <TextInput
                    label="Disabled Outlined TextInput"
                    mode="outlined"
                    value={disabledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    outlineColor={theme.colors.outline}
                    onChangeText={(value) => setDisabledText(value)}
                    disabled
                    style={styles.outlineInput}
                />
            </View>

            <View style={styles.container}>
                <Text> Outline Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalFilledText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalFilledText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Outline Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorFilledText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorFilledText(value);
                        setHasError(value.length > 4);
                    }}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Outline Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    value={disabledFilledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledFilledText(value)}
                    disabled
                    style={[styles.flatInput]}
                />
            </View>
            <View style={styles.container}>
                <Text> Dense Outline TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalText}
                    dense={true}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Dense Outline TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorText(value);
                        setHasError(value.length > 5);
                    }}
                    dense={true}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Dense Outline TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    value={disabledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledText(value)}
                    disabled
                    dense={true}
                    style={[styles.flatInput]}
                />
            </View>
            <View style={styles.container}>
                <Text> Dense Outline Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={normalFilledText}
                    dense={true}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setNormalFilledText(value)}
                    style={[styles.flatInput]}
                />

                <Text> Error Dense Outline Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    value={errorFilledText}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => {
                        setErrorFilledText(value);
                        setHasError(value.length > 4);
                    }}
                    dense={true}
                    error={hasError}
                    style={{ marginHorizontal: 8, marginTop: 8 }}
                />
                <HelperText type="error" visible={hasError} style={{ marginBottom: 8 }}>
                    Error Message
                </HelperText>

                <Text> Disabled Dense Outline Filled TextInput </Text>
                <TextInput
                    label="TextInput"
                    mode="outlined"
                    value={disabledFilledText}
                    left={<TextInput.Icon icon="email" />}
                    right={<TextInput.Icon icon="menu-down" />}
                    underlineColor={theme.colors.onSurfaceVariant}
                    onChangeText={(value) => setDisabledFilledText(value)}
                    disabled
                    dense={true}
                    style={[styles.flatInput]}
                />
            </View>
        </View>
    );
};
