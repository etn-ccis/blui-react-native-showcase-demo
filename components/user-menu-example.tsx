import { IconFamily, InfoListItemProps, UserMenu } from '@brightlayer-ui/react-native-components';
import React from 'react';
import { Avatar } from 'react-native-paper';
import * as Colors from '@brightlayer-ui/colors';

const SwapIcon: IconFamily = {
    family: 'material',
    name: 'swap-horiz',
    direction: 'ltr',
};
const InvertColorsIcon: IconFamily = {
    family: 'material',
    name: 'invert-colors',
    direction: 'ltr',
};
const CancelIcon: IconFamily = {
    family: 'material',
    name: 'cancel',
    direction: 'ltr',
};

type UserMenuExampleProps = {
    onToggleRTL: () => void;
    onToggleTheme: () => void;
};

export const UserMenuExample: React.FC<UserMenuExampleProps> = (props) => {
    const { onToggleRTL, onToggleTheme } = props;

    const menuItems: InfoListItemProps[] = [
        { title: 'Toggle RTL', icon: SwapIcon, onPress: (): void => onToggleRTL() },
        {
            title: 'Toggle Theme',
            icon: InvertColorsIcon,
            onPress: (): void => onToggleTheme(),
        },
        { title: 'Cancel', icon: CancelIcon },
    ];

    return (
        <UserMenu
            menuTitle={'John Smith'}
            menuSubtitle={'Account Manager'}
            menuItems={menuItems}
            avatar={
                <Avatar.Icon
                    icon="account-circle"
                    size={40}
                    color={Colors.blue[500]}
                    style={{ backgroundColor: Colors.blue[50] }}
                />
            }
        />
    );
};
