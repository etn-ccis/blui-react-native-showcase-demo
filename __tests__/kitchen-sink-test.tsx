import 'react-native';
import React from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { KitchenSink } from '../components/kitchen-sink';
import renderer from 'react-test-renderer';

describe('Kitchen sink snapshot', () => {
    test('Renders correctly and snapShot matches', ()=>{
    const snap = renderer.create(
        <SafeAreaProvider>
        <KitchenSink />
        </SafeAreaProvider>
    ).toJSON();
    expect(snap).toMatchSnapshot();
})
});
