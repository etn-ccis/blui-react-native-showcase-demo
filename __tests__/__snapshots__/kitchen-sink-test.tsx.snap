// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kitchen sink snapshot Renders correctly and snapShot matches 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
